resource/wrappedobject.go

```diff
@@ -0,0 +1,320 @@
+
+func (o *WrappedObject[T]) GetDeletionGracePeriodSeconds() *int64 {
+	return o.Object.GetDeletionGracePeriodSeconds()
+}
+
+func (o *WrappedObject[T]) SetDeletionGracePeriodSeconds(i *int64) {
+	o.Object.SetDeletionGracePeriodSeconds(i)
+}
+
+func (o *WrappedObject[T]) GetLabels() map[string]string {
+	return o.Object.GetLabels()
+}
+
+func (o *WrappedObject[T]) SetLabels(labels map[string]string) {
+	o.Object.SetLabels(labels)
+}
+
+func (o *WrappedObject[T]) GetAnnotations() map[string]string {
+	return o.Object.GetAnnotations()
+}
+
+func (o *WrappedObject[T]) SetAnnotations(annotations map[string]string) {
+	o.Object.SetAnnotations(annotations)
+}
+
+func (o *WrappedObject[T]) GetFinalizers() []string {
+	return o.Object.GetFinalizers()
+}
+
+func (o *WrappedObject[T]) SetFinalizers(finalizers []string) {
+	o.Object.SetFinalizers(finalizers)
+}
+
+func (o *WrappedObject[T]) GetOwnerReferences() []metav1.OwnerReference {
+	return o.Object.GetOwnerReferences()
+}
+
+func (o *WrappedObject[T]) SetOwnerReferences(references []metav1.OwnerReference) {
+	o.Object.SetOwnerReferences(references)
+}
+
+func (o *WrappedObject[T]) GetManagedFields() []metav1.ManagedFieldsEntry {
+	return o.Object.GetManagedFields()
+}
+
+func (o *WrappedObject[T]) SetManagedFields(managedFields []metav1.ManagedFieldsEntry) {
+	o.Object.SetManagedFields(managedFields)
+}
+
+func (*WrappedObject[T]) GetSpec() any {
+	return nil
+}
+
+func (*WrappedObject[T]) SetSpec(_ any) error {
+	return fmt.Errorf("no spec to set in wrapped object")
+}
+
+func (*WrappedObject[T]) GetSubresources() map[string]any {
+	return map[string]any{}
+}
+
+func (*WrappedObject[T]) GetSubresource(_ string) (any, bool) {
+	return nil, false
+}
+
+func (*WrappedObject[T]) SetSubresource(_ string, _ any) error {
+	return fmt.Errorf("cannot set subresource in wrapped object")
+}
+
+func (o *WrappedObject[T]) GetStaticMetadata() StaticMetadata {
+	return StaticMetadata{
+		Name:      o.Object.GetName(),
+		Namespace: o.Object.GetNamespace(),
+		Group:     o.Object.GroupVersionKind().Group,
+		Version:   o.Object.GroupVersionKind().Version,
+		Kind:      o.Object.GroupVersionKind().Kind,
+	}
+}
+
+func (o *WrappedObject[T]) SetStaticMetadata(metadata StaticMetadata) {
+	o.Object.SetName(metadata.Name)
+	o.Object.SetNamespace(metadata.Namespace)
+	o.Object.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   metadata.Group,
+		Version: metadata.Version,
+		Kind:    metadata.Kind,
+	})
+}
+
+//nolint:revive,staticcheck
+func (o *WrappedObject[T]) GetCommonMetadata() CommonMetadata {
+	var err error
+	dt := o.Object.GetDeletionTimestamp()
+	var deletionTimestamp *time.Time
+	if dt != nil {
+		deletionTimestamp = &dt.Time
+	}
+	updt := time.Time{}
+	createdBy := ""
+	updatedBy := ""
+	annotations := o.Object.GetAnnotations()
+	if annotations != nil {
+		strUpdt, ok := annotations[AnnotationUpdateTimestamp]
+		if ok {
+			updt, err = time.Parse(time.RFC3339, strUpdt)
+			if err != nil {
+				// HMMMM
```

Comment:
We can try another format as a fallback here if there's any other probable format, but I guess in the end we have to ignore the error.